cli/download_source.go

```diff
@@ -158,3 +165,32 @@ func downloadSource(terraformSource *tfsource.TerraformSource, terragruntOptions
 
 	return nil
 }
+
+// Check if working terraformSource.WorkingDir exists and is directory
+func validateWorkingDir(terraformSource *tfsource.TerraformSource) error {
+	fileInfo, err := os.Stat(terraformSource.WorkingDir)
+	if err != nil {
+		return WorkingDirNotFound{terraformSource.WorkingDir}
+	}
+
+	if !fileInfo.IsDir() {
+		return WorkingDirNotDir{terraformSource.WorkingDir}
+	}
+	return nil
+}
+
+type WorkingDirNotFound struct {
+	Dir string
+}
+
+func (err WorkingDirNotFound) Error() string {
+	return fmt.Sprintf("Directory not found: %s", err.Dir)
```

Comment:
We may want a more specific error message that includes the `source` URL. E.g., "Working dir XXX from source URL YYY does not exist."